### DISCUSSION TITLE
How to update Plex LXC to unprivilaged and copy old Plex data?

### DISCUSSION THREAD

**Discussion by di5cord20** [tteck/Proxmox#2867] [Help Needed] [answered]
I opted to create a new unprivileged Plex LXC as suggested here (https://github.com/tteck/Proxmox/discussions/2862#discussioncomment-9241951). Once completed, I verified that Plex was running (didn't sign in or go beyond initial splash screen). I then immediately ran the copy script (per https://github.com/tteck/Proxmox/discussions/1762#discussioncomment-6916106) which completed successfully. Unfortunately, when I go back to the new Plex instance (IP:32400/web), I'm met with a "This site can’t be reached" error.

Any ideas?

**Comment by tteck** [comment]
The mentioned script is coping `/var/lib/plexmediaserver/Library/` to `/var/lib/plexmediaserver/Library/` so, I'm unaware of anything that would cause an error.

**Reply by tteck** [reply]
Maybe the new Plex LXC needs to be set up first? I can't remember.

**Reply by di5cord20** [reply]
I'll give that a try and report back

**Reply by di5cord20** [reply]
Unfortunately I got the same result.

- setup Plex first by
- signed in with my plex pass account
- setup new server, didn't setup any libraries
- ran copy script (which completed successfully
- started the new unprivilaged plex lxc
- got the same error as before ("This site can’t be reached")

**Reply by di5cord20** [reply]
Update: just realized I don't use the default 32400 port.  I'm going to try again and use my custom port after the copy.  Will report back soon.

**Reply by di5cord20** [reply]
Cancel that, the non-default port is only when accessing externally.

**Reply by di5cord20** [reply]
I tested to confirm if the data copied successfully. I noticed the permissions/group are different. Could that be the issue?

Source:
![image](https://github.com/tteck/Proxmox/assets/140566995/fea9bb80-2578-4a82-9085-19bda2d981b1)
```
root@plex:/var/lib/plexmediaserver# ls -la
total 12
drwxr-xr-x  3 plex plex 4096 Feb 10 19:55 .
drwxr-xr-x 25 root root 4096 Feb 10 19:55 ..
drwxr-xr-x  3 plex plex 4096 Feb 10 19:55 Library
```

Destination:
![image](https://github.com/tteck/Proxmox/assets/140566995/33e3329f-9d35-4ee3-8688-e272ba7543bb)
```
root@plex:/var/lib/plexmediaserver# ls -la
total 12
drwxr-xr-x  3 plex   plex    4096 Apr 26 21:33 .
drwxr-xr-x 25 root   root    4096 Apr 26 21:33 ..
drwxr-xr-x  3 nobody nogroup 4096 Feb 10 19:55 Library
```

**Reply by tteck** [reply]
you can try `chown -R plex:plex /var/lib/plexmediaserver/Library/`

**Reply by di5cord20** [reply]
I get an "operation not permitted" error when I try to chown

**Reply by tteck** [reply]
`chown -R plex:plex /var/lib/plexmediaserver/`?

**Reply by di5cord20** [reply]
plexmediaserver is already owned by plex:plex on the destination.  It starts to diverge at the Library folder.

**Reply by tteck** [reply]
🤷🏻‍♂️ don't know, other than starting fresh with a new unprivileged Plex LXC. Both Ubuntu 22.04?

**Reply by tteck** [reply]
`chown -R plex:plex /var/lib/plexmediaserver/*`?

**Reply by di5cord20** [reply]
Yup, I started with a new unprivileged Plex LXC using Ubuntu 22.04.  I go the Advanced route for all the LXC installs when I use your amazing script so that I can select the container ID, modify the default disk/ram, and map specific IPs (I like the IP to match the container ID).

**Reply by di5cord20** [reply]
Same error on `chown -R plex:plex /var/lib/plexmediaserver/*` ... "operation not permitted"

**Reply by tteck** [reply]
I'm out of ideas, sorry.

**Reply by tteck** [reply]
One last thing to try

```
shopt -s dotglob
chown -R plex:plex /var/lib/plexmediaserver/Library/
```

**Reply by di5cord20** [reply]
Appreciate all your help (as always).  Unfortunately the 

```
shopt -s dotglob
chown -R plex:plex /var/lib/plexmediaserver/Library/
```

didn't do the trick either.

**Reply by di5cord20** [reply]
I googled the issue specific to proxmox and came across https://forum.proxmox.com/threads/unprivileged-lxc-user-mapping-getting-operation-not-permitted-for-chown.126649/.  Following the thread, I mounted the LXC on the host and tried to chown the directory.  Unfortunately, the plex user/group doesn't exist on the host.  I then tried with 999:999 instead (as that the id:group from the container - found via `cat /etc/passwd`.  The command ran without errors but didn't change the group:

```
root@pve:/var/lib/lxc/215/rootfs/var/lib/plexmediaserver# ls -las
total 12
4 drwxr-xr-x  3 100999          100999 4096 Apr 26 21:33 .
4 drwxr-xr-x 25 100000          100000 4096 Apr 26 21:33 ..
4 drwxr-xr-x  3    999 systemd-journal 4096 Feb 10 19:55 Library
```

In case it helps, here's what the destination plex lxc shows for `cat /etc/passwd`
`plex:x:999:999::/var/lib/plexmediaserver:/usr/sbin/nologin`

The host doesn't have a plex or 999 id:group.  

I'm still stumped... but haven't given up yet :)

**Reply by tteck** [reply]
Before making many nasty modifications, I'd suggest starting fresh with a new unprivileged Plex LXC if possible. https://github.com/tteck/Proxmox/discussions/2850#discussioncomment-9242753

**Reply by di5cord20** [reply]
Success!!!! 
I noticed that 100999 is the id:group on the "." item (which is plex:plex on the lxc itself).  So I did a `chown -R 100999:100999 Library/` on the mounted LXC in the host ... and it worked.  When I then checked in the LXC console directly, the Library folder is now owned by plex:plex.  I proceeded to unmount the LXC filesystem from the host, then rebooted the new plex destination LXC (just to be safe).  I then navigated to the new plex unprivileged lxc IP:32400/web and it worked !!!

@tteck - I couldn't have solved without some of the troubleshooting steps you provided.  They led me on the right path.  Thanks again.

For the benefit of the community, not sure if the plex copy script needs to be updated so that the Library folder permissions are correct.  If that's not possible, the manual steps I took could be a good workaround.  I'm happy to document them again if it would help (while it's still fresh in my mind).

**Reply by tteck** [reply]
Just a note, the plex copy script was designed to be used from a privileged to a privileged container. No doubt permissions differ from a privileged to an unprivileged container.

**Reply by di5cord20** [reply]
That makes sense.  I ended up writing out my manual steps below so others could reference if needed.

**Comment by di5cord20** [comment, marked as answer]
Here's my steps to manually migrate plex from privilaged (source) to new unprivileged (destination):

1. Install Plex LXC using the updated tteck script (`bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/ct/plex.sh)"`).  Make sure to select Ubuntu 22.04 and Unprivileged.
2. Once installed, use tteck's copy script (originalled discussed https://github.com/tteck/Proxmox/discussions/1762#discussioncomment-6916106) to copy the source to destination
3. Here's the tricky part, after the copy completed, you'll need to mount the destination LXC to your host so that you can update permissions for the Plex Libary folder.
4. To mount the destination on your host, open the host shell and run `pct mount <destination lxc id>` (replace <destination lxc id> with you container id)
5. Once mounted, navigate to `/var/lib/lxc/<destination lxc id>/rootfs/var/lib/plexmediaserver` and perform an `ls -la` to confirm id:group ownership (in my case, it was 999:systemd-journal).  You'll need to modify ownership to plex:plex .  The problem is the plex:plex id:group doesn't exist on the host.  To solve, use 100999:100999 instead by running `chown -R 100999:100999 Library`
6. Confirm the permissions updated via `ls-la`
7. Unmount the destination LXC from the host via `pct unmount <destination lxc id>`
8. Reboot the destination LXC

That's it.  At this point, the /var/lib/plexmediaserver/Library folder permissions should be plex:plex and the new destination unprivileged Plex LXC should now be working.

Hope this helps others.  @tteck - thanks again for the wonderful scripts and fabulous support!  Couldn't have figured this out without your help.

**Reply by tteck** [reply]
Here's the full list of the "copy data to another LXC" scripts: https://github.com/tteck/Proxmox/tree/main/misc/copy-data
